### PR TITLE
Upgrade xorm to new version which supported update join for all supported databases

### PR DESCRIPTION
**Pull request by lunny**
Fix https://github.com/go-gitea/gitea/pull/28547#issuecomment-1867740842

Since https://gitea.com/xorm/xorm/pulls/2383 merged, xorm now supports UPDATE JOIN.
To keep consistent from different databases, xorm use `engine.Join().Update`, but the actural generated SQL are different between different databases.

For MySQL, it's `UPDATE talbe1 JOIN table2 ON join_conditions SET xxx Where xxx`.

For MSSQL, it's `UPDATE table1 SET xxx FROM TABLE1, TABLE2 WHERE join_conditions`.

For SQLITE per https://www.sqlite.org/lang_update.html, sqlite support `UPDATE table1 SET xxx FROM table2 WHERE join conditions` from 3.33.0(2020-8-14).

POSTGRES is the same as SQLITE.

### CODE REVIEW

**Review by KN4CK3R — APPROVED** [review]

File: models/issues/comment.go [hunk 1]
```diff
@@ -1161,14 +1161,9 @@ func DeleteComment(ctx context.Context, comment *Comment) error {
...
-				InnerJoin("repository", "issue.repo_id = repository.id").
-				Where(builder.Eq{
-					"repository.original_service_type": tp,
-				}),
-		)).
+		Join("INNER", "issue", "issue.id = comment.issue_id").
+		Join("INNER", "repository", "issue.repo_id = repository.id").
+		Where("repository.original_service_type = ?", tp).
 		And("comment.original_author_id = ?", originalAuthorID).
 		Update(map[string]any{
```
Comment (delvh) [inline models/issues/comment.go:1168]: I thought SQLite does not support `UPDATE JOIN`?
Or does xorm automatically generate the correct query?

**Review by delvh — APPROVED** [review]

File: models/issues/comment.go (hunk 1 above)
Comment (delvh) [reply models/issues/comment.go:1168]: Or was it only `xorm does not support UPDATE JOIN on SQLite`?

Comment (6543) [reply models/issues/comment.go:1168]: https://gitea.com/xorm/xorm/pulls/2383 looks like it was just blocked by xorm

**Review by 6543 — APPROVED** [review]

File: models/issues/comment.go (hunk 1 above)
Comment (delvh) [reply models/issues/comment.go:1168]: Yes, but does this work under SQLite?
I've just looked through that PR, and to me, it looks like for SQLite, still an `UPDATE JOIN` is generated, even though that is not supported AFAIK.

Comment (6543) [reply models/issues/comment.go:1168]: https://www.sqlite.org/lang_update.html i would say as long as you have the join-select within the where it works - through the changes of xorm dont reflect that yes

tests pass ?!? dont we have coverage of it?!?

Comment (lunny) [reply models/issues/comment.go:1168]: per https://www.sqlite.org/lang_update.html, sqlite support `UPDATE table1 SET xxx FROM table2 WHERE join conditions` from 3.33.0(2020-8-14).

**Pull request merged**